app/uk/gov/hmrc/ngrraldfrontend/controllers/AgreementController.scala

```diff
@@ -133,22 +135,36 @@ class AgreementController @Inject()(view: AgreementView,
   }
 
 
-  def show: Action[AnyContent] = {
-    (authenticate andThen hasLinkedProperties).async { implicit request =>
-      request.propertyLinking.map(property =>
-        Future.successful(Ok(view(
-          selectedPropertyAddress = property.addressFull,
-          form,
-          dateInput(),
-          buildRadios(form, openEndedRadio(form)),
-          buildRadios(form, breakClauseRadio(form))
-        )))
-      ).getOrElse(throw new NotFoundException("Couldn't find property in mongo"))
+  def show(mode: Mode): Action[AnyContent] = {
+    (authenticate andThen getData).async { implicit request =>
+      val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.credId)).get(AgreementPage) match {
+        case None => form
+        case Some(value) => form.fill(AgreementForm(NGRDate.fromString(value.agreementStart), if (value.isOpenEnded) {
+          "YesOpenEnded"
+        } else {
+          "NoOpenEnded"
+        }, value.openEndedDate match {
+          case Some(value) => Some(NGRDate.fromString(value))
+          case None => None
+        }, if (value.haveBreakClause) {
```

Comment:
I'm thinking storing things as boolean is making our code more annoying then good, we should raise this.

Reply:
Yeah I think in the future we should have some kind of standard. A lot of the radio buttons are strings which are just simple Yes or No answers. So I think we should be storing them as booleans as it makes it easier to prepopulate the form rather than having to put in the name of the radio value. If you get me. For example what your rent includes page really should just be booleans for each value not the whole value that you have to pass back